active_learning/tool_tracking_al/active_learning.py

```diff
@@ -0,0 +1,573 @@
+    RandZoomd,
+    Resized,
+    ToMetaTensord,
+    Compose,
+    AsDiscrete,
+)
+
+from torch.utils.tensorboard import SummaryWriter
+from monai.networks.nets import FlexibleUNet
+
+# Argument Parser
+parser = argparse.ArgumentParser(description="Active Learning Settings")
+
+# Directory & Json & Seed
+parser.add_argument("--base_dir", default="/home/vishwesh/experiments/robo_tool_experiments/variance_sanity", type=str)
+parser.add_argument("--data_root", default="/scratch_2/robo_tool_dataset_2023", type=str)
+parser.add_argument("--json_path", default="/scratch_2/robo_tool_dataset_2023/data_list.json", type=str)
+parser.add_argument("--seed", default=120, type=int)
+
+# Active learning parameters
+parser.add_argument("--active_iters", default=3, type=int)
+parser.add_argument("--dropout_ratio", default=0.2, type=float)
+parser.add_argument("--mc_number", default=10, type=int)
+parser.add_argument("--initial_pool", default=10, type=int)
+parser.add_argument("--queries", default=10, type=int)
+parser.add_argument("--strategy", default="variance", type=str)  # Options available variance, random
+
+# DL Hyper-parameters
+parser.add_argument("--steps", default=300, type=int)  # Default is 2000, this needs to be tweaked more
+parser.add_argument("--val_interval", default=1, type=int)
+parser.add_argument("--batch_size", default=1, type=int)
+parser.add_argument("--val_batch_size", default=1, type=int)
+parser.add_argument("--lr", default=1e-4, type=float)
+
+
+def main():
+    # Argument parser Code    # monai.config.print_config()
+    args = parser.parse_args()
+
+    print("MAIN Argument values:")
+    for k, v in vars(args).items():
+        print(k, "=>", v)
+        print("-----------------")
+
+    # Set Determinism
+    set_determinism(seed=args.seed)
+
+    # Base directory where all model collections will go
+    base_model_dir = os.path.normpath(args.base_dir)
+    if os.path.exists(args.base_dir) == False:
+        os.mkdir(args.base_dir)
+
+    base_model_dir = os.path.join(base_model_dir, "all_models")
+    if os.path.exists(base_model_dir) == False:
+        os.mkdir(base_model_dir)
+
+    json_base_dir = os.path.join(base_model_dir, "all_jsons")
+    if os.path.exists(json_base_dir) == False:
+        os.mkdir(json_base_dir)
+
+    fig_base_dir = os.path.join(base_model_dir, "qa_figs")
+    if os.path.exists(fig_base_dir) == False:
+        os.mkdir(fig_base_dir)
+
+    # Root data path
+    data_root = os.path.normpath(args.data_root)
+
+    # Load Json and append root path
+    with open(args.json_path, "r") as json_f:
+        p_json_data = json.load(json_f)
+
+    p_train_data = p_json_data["train"]
+    p_copy_train_data = copy.deepcopy(p_train_data)
+    p_val_data = p_json_data["validation"]
+    p_copy_val_data = copy.deepcopy(p_val_data)
+    p_test_data = p_json_data["test"]
+    p_copy_test_data = copy.deepcopy(p_test_data)
+
+    print("Random Strategy is being used for selection instead of SSL based ranking ...")
```

Comment:
Please confirm the random strategy is always applied as I see other options like "variance"

Reply:
Yes for initial data selection random is always used, irrespective of any given strategy.